modules/base-realms/realm-github/realm.tf

```diff
@@ -0,0 +1,19 @@
+module "realm" {
+  source     = "../../realm"
+  realm_name = var.realm_name
+}
+
+data "keycloak_authentication_execution" "browser_identity_provider_redirector" {
+  realm_id          = module.realm.id
```

Comment:
out of curiosity why are things like realm name passed in through `var` but id's passed in through the module?  Is it because the id is generated by terraform or keycloak?